### PR TITLE
Update release pipeline

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #4961

* Publish insiders daily at 2am PDT (9am UTC)
* Publish release

### CODE REVIEW

File: .github/workflows/build-test.yml
Comment (DonJayamanne) [inline .github/workflows/build-test.yml:76]: We push insiders to marketplace only during the nightly run

File: .github/workflows/build-test.yml
Comment (DonJayamanne) [inline .github/workflows/build-test.yml:88]: We push insiders to marketplace only during push to release & commits has special comments

File: .github/workflows/build-test.yml
Comment (DonJayamanne) [inline .github/workflows/build-test.yml:1070]: Once reviewed, i'll enable this, only way to find out is to push a VSIX for insiders and test.

File: .github/workflows/build-test.yml
Comment (DonJayamanne) [inline .github/workflows/build-test.yml:1070]: Publish insider/release

File: .github/workflows/build-test.yml
Comment (DonJayamanne) [inline .github/workflows/build-test.yml:1073]: All of the rest only applies when we generate stable releases

File: .github/workflows/build-test.yml
Comment (DonJayamanne) [inline .github/workflows/build-test.yml:23]: Nightly job

File: .github/workflows/build-test.yml
Comment (DonJayamanne) [inline .github/workflows/build-test.yml:1042]: Using Github Environments Releases
I have created two environments `insider` & `stable`.
For Insiders, they will get published directly.
For `stable` environment, this job will NOT run, until it is approved. there is a list of approvers that we can add (for now I had only added myself, to ensure this works properly, then we can add everyone else).

See here for more info:
https://docs.github.com/en/actions/reference/environments
https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments

**Review by rchiodo — APPROVED** [review]

File: .github/workflows/build-test.yml
Comment (DonJayamanne) [reply .github/workflows/build-test.yml:1042]: Oh yes, I will add @IanMatthewHuff  as he's going to be doing the release this time

**Review by DavidKutu — APPROVED** [review]

**Pull request merged**
